### PR TITLE
  ⚡️(front) optimize streaming markdown rendering with block splitting

### PR DESCRIPTION
**Pull request by providenz**
Improve frontend performance when handling streaming content.

## Purpose

Instead of re-rendering the entire message through  markdown on every stream update, render completed paragraphs as markdown (+ code highlighting and math formatting) once   and freeze them, while streaming new content as plain text until it forms a   complete block rady to be rendred.

cf. https://github.com/orgs/suitenumerique/projects/13/views/3?pane=issue&itemId=142314043&issue=suitenumerique%7Cconversations%7C183

## Proposal

  Split streaming content into independent blocks to avoid re-rendering  all markdown on each stream update. Before, the entire content was re-parsed through markdown/katex/syntax-highlighting on every chunk updates.

  - Split content into completedBlocks list instead of single string
  - Add BlockList memoized component for independent block rendering
  - Each completed block is rendered once and never re-processed
  - Only pending (incomplete) content is rendered as raw text
  - Fix splitStreamingContent edge case with trailing newlines
  - Use stable message keys to prevent unnecessary remounts
  - Add tests for MessageBlock and MessageItem components
  -  Stabilize openSources callback with functional setState (empty deps)
  - Add custom arePropsEqual to skip re-renders for non-streaming messages
  - Move markdown plugins (shiki, katex, remarkGfm) to module-level singletons
  - Extract MessageItems and MessageBlocks from Chas.tsx for granular memoization
  - Memoize derived values (last and first messages)

## Summary by CodeRabbit

* **New Features**
  * Streaming-friendly message rendering with progressive markdown and raw-text handling, plus copy, attachments, source toggles, and improved accessibility.

* **Performance**
  * Front-end optimizations for faster, smoother streaming markdown rendering.

* **Refactor**
  * Message rendering moved into a dedicated component for clearer structure and reduced re-renders.

* **Tests**
  * Extensive tests covering streaming, block-splitting, interactions, and edge cases.

* **Chores**
  * Changelog updated with an unreleased "Changed" entry.

### CODE REVIEW

File: src/frontend/apps/conversations/src/features/chat/components/MessageItem.tsx [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+  }
+  if (prevProps.isMobile !== nextProps.isMobile) {
+    return false;
+  }
+  if (prevProps.conversationId !== nextProps.conversationId) {
+    return false;
+  }
+
+  return true;
+};
```
[the remark this answers is not in the document]
Comment (providenz) [reply src/frontend/apps/conversations/src/features/chat/components/MessageItem.tsx]: getMetadata is defined with useCallback(() => metadataCache.get(url), []) — empty
  deps, so the function reference is created once and never changes. Non need to compare
